modules/api/src/main/java/net/countercraft/movecraft/events/CraftCollisionEvent.java

```diff
@@ -3,23 +3,32 @@
 import net.countercraft.movecraft.craft.Craft;
 import net.countercraft.movecraft.utils.BitmapHitBox;
 import net.countercraft.movecraft.utils.HashHitBox;
+
+import org.bukkit.World;
 import org.bukkit.event.HandlerList;
 import org.jetbrains.annotations.NotNull;
 
 public class CraftCollisionEvent extends CraftEvent{
     private static final HandlerList HANDLERS = new HandlerList();
     @NotNull private final BitmapHitBox hitBox;
+    @NotNull private final World world;
     private boolean isCancelled = false;
 
-    public CraftCollisionEvent(@NotNull Craft craft, @NotNull BitmapHitBox hitBox) {
+    public CraftCollisionEvent(@NotNull Craft craft, @NotNull BitmapHitBox hitBox, @NotNull World world) {
```

Comment:
World shouldn't need to be specified since its not modified here and should already exist in the craft (thus there should also not be a getter).

Reply:
The block that the craft is colliding with could be in a different world if the translation is cross-world